mx.h

```diff
@@ -250,6 +298,9 @@ struct MxOps
    * @param buf Buffer containing tags
    * @retval  0 Success
    * @retval -1 Failure
+   *
+   * **Contract**
+   * - @a m is not NULL
    */
```

Comment:
What about `e` and `buf`?
Also, we could check that `buf` isn't empty (`buf[0] != '\0')
See check in `nm_tags_commit()`

Reply:
handled the !e and !buf cases.... but we can't use an empty buffer as a failure. The imap worker uses the absence of any tags to clear all flags (iiuc).

As in, it doesn't have notmuch +/-tag semantics but rather just replaces all tags with whatever the user was typing, including ""

Or do I misunderstand how this works?

Reply:
you're right, we don't want the (buf[0] != '\0') check